packages/serverless/src/domain/use-case/collection.ts

```diff
@@ -173,22 +165,22 @@ async function dropIndex(
   databaseName: string,
   actor: string,
   collectionName: string,
-  indexName: string,
-  session?: ClientSession
+  indexName: string
 ): Promise<boolean> {
   if (
-    await hasCollectionPermission(
-      databaseName,
-      collectionName,
-      actor,
-      'system',
-      session
-    )
+    await hasCollectionPermission(databaseName, collectionName, actor, 'system')
   ) {
-    // TODO: Should we check if index fields exist for a collection
-    return ModelCollection.getInstance(databaseName, collectionName).dropIndex(
-      indexName,
-      { session }
+    const index = `${indexName}_1`
+    if (await doesIndexExist(databaseName, actor, collectionName, index)) {
+      // TODO: Should we check if index fields exist for a collection
```

Comment:
If you already check. You should remove the `TODO:`

Reply:
removed